### PR TITLE
crypto/pow: add Bitcoin-style nBits proof-of-work (i663)

### PR DESCRIPTION
**Pull request by jose-compu**
## Summary

Implements [i663-crypto-pow](./issues/663-crypto-pow.md) — closes GitHub [#930](https://github.com/functionalscript/functionalscript/issues/930).

- Add `fs/crypto/pow/` with `targetFromNBits`, `pow(sha2)`, and genesis constants.
- **nBits** decoding: `target = mantissa × 2^(8 × (exponent − 3))`; reject negative, overflow, and >256-bit targets (Bitcoin `SetCompact` rules).
- **PoW check:** `hashInt(data) ≤ target` via injected `Sha2` (`computeSync` + `uint` from `fs/crypto/sha2`).
- Export `./fs/crypto/pow/module.f.ts` in `deno.json`; document in `fs/crypto/pow/README.md` and `fs/crypto/README.md`.
- **21 proofs** in `proof.f.ts`: genesis + block #0 target, compact edge cases, `meets`/`verify`, SHA-224/256 `hashInt`.

Mining/search loops stay out of scope (callers iterate nonces and call `meets`).

## Test plan

- [x] `npx tsc`
- [x] `npm run fst` in `fs/crypto/pow/` (21/21)
- [ ] `npm test` (full suite)
- [ ] `npm run update` (if not run on CI)

### CODE REVIEW

File: fs/crypto/pow/module.f.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    return target
+}
+
+export type Pow = {
+    /** Hash `data` with the configured `Sha2`; digest as big-endian uint256. */
+    readonly hashInt: (data: Vec) => bigint
+    /** Whether `hashInt(data) <= targetFromNBits(nBits)`. */
+    readonly meets: (nBits: bigint) => (data: Vec) => boolean
+    /** Same as {@link Pow.meets}. */
+    readonly verify: (nBits: bigint) => (data: Vec) => boolean
```
Comment (sergey-shandar) [inline fs/crypto/pow/module.f.ts]: Why do we need two functions with the same functionality?

File: fs/crypto/pow/module.f.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+ *
+ * @param hash - SHA-2 configuration whose digest is compared as uint256.
+ */
+export const pow = (hash: Sha2): Pow => {
+    const c = computeSync(hash)
+    const hashInt = (data: Vec): bigint => uint(c([data]))
+    const meets = (nBits: bigint) => (data: Vec): boolean =>
+        hashInt(data) <= targetFromNBits(nBits)
+    return { hashInt, meets, verify: meets }
+}
```
Comment (sergey-shandar) [inline fs/crypto/pow/module.f.ts:82]: We may also export popular PoWs, for example, `sha256Pow = pow(sha256)`, or `bitcoinPoW`.

File: fs/crypto/pow/module.f.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+ * @param nBits - Compact target encoding.
+ * @returns Decoded target as a big-endian unsigned integer.
+ */
+export const targetFromNBits = (nBits: bigint): bigint => {
+    const exponent = nBits >> exponentShift
+    const mantissa = nBits & nBitsMantissa
+    if (negativeNBits(mantissa)) { throw 'negative nBits' }
+    const target = compactTarget(exponent)(mantissa)
+    if (overflowNBits(exponent)(mantissa)(target)) { throw 'overflow nBits' }
+    if (target > uint256Mask) { throw 'target exceeds 256 bits' }
```
Comment (sergey-shandar) [inline fs/crypto/pow/module.f.ts]: FunctionalScript uses `throw` as `panic` in Rust. If the error is not critical, use `Result<bigint, ...>`, `Nullable<bigint>`, or `Option<bigint>` instead.

**Review by sergey-shandar — COMMENTED** [review]
LGTM, there are a few comments, mostly questions and suggestions.

File: fs/crypto/pow/module.f.ts (hunk 1 above)
Comment (jose-compu) [reply fs/crypto/pow/module.f.ts]: ok, fixed here: https://github.com/functionalscript/functionalscript/pull/935/changes/d20ae29be9fb001ef97d8acd616f9d1b3bc86fe8

File: fs/crypto/pow/module.f.ts (hunk 2 above)
Comment (jose-compu) [reply fs/crypto/pow/module.f.ts:82]: added here : https://github.com/functionalscript/functionalscript/pull/935/changes/c207e8f1125af3c02bd88f838b5a67276d657dc2

File: fs/crypto/pow/module.f.ts (hunk 3 above)
Comment (jose-compu) [reply fs/crypto/pow/module.f.ts]: Refactored per review: `targetFromNBits` now returns `Nullable<bigint>` (`null` on invalid nBits, no `throw`). `meets` returns `false` when decode is `null`. Proofs updated; added `meets.invalidNBits`.

See here: https://github.com/functionalscript/functionalscript/pull/935/changes/8fcf4b24d46cdf5da906f1f43daeeb5a48f222cd?notification_referrer_id=NT_kwHOAdQu-9oAJVJlcG9zaXRvcnk7NDIwMDQxNDcxO0lzc3VlOzQ1ODYzNzEwNTc&notifications_query=is%3Aunread

**Review by sergey-shandar — APPROVED** [review]
